File: 1582.cpp
struct Solution {
    int numSpecial(vector<vector<int>>& mat) {
        int m = mat.size(), n = mat[0].size();
        vector<int> r(m, 0), c(n, 0);
        for (int i = 0; i < m; ++i){
            for (int j = 0; j < n; ++j){
                if (mat[i][j]){
                    ++r[i]; ++c[j];
                }
            }
        }
        int counter = 0;
        for (int i = 0; i < m; ++i){
            if (r[i] != 1){
                continue;
            }
            for (int j = 0; j < n; ++j){
                if (c[j] == 1){
                    counter += mat[i][j];
                }
            }
        }
        return counter;
    }
};

File: sol/1582.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        m, n = len(mat), len(mat[0])
        r = [0 for _ in range(m)]
        c = [0 for _ in range(n)]
        reg = list()
        for i in range(m):
            for j in range(n):
                if mat[i][j]:
                    r[i] += 1
                    c[j] += 1
                    reg.append((i, j))
        return sum(1 for (i, j) in reg if r[i] == c[j] == 1)

